src/controllers/dynakube/activegate/internal/capability/reconciler.go

```diff
@@ -42,11 +42,12 @@ func (r *Reconciler) Reconcile() error {
 	if err != nil {
 		return errors.WithStack(err)
 	}
-
-	if r.dynakube.NeedsActiveGateService() {
-		err = r.createOrUpdateService()
-		if err != nil {
-			return errors.WithStack(err)
+	if !r.capability.AssistsSynthetic() {
```

Comment:
Could be 
```suggestion
	if !r.capability.AssistsSynthetic() && r.dynakube.NeedsActiveGateService() { 
	...
	}
```
to limit the depth

also the `NeedsActiveGateService` could be updated to check the synthetic feature flag

Reply:
Rephrasing.